pkg/api/kyverno/v1/types.go

```diff
@@ -188,9 +188,8 @@ type PolicyViolationSpec struct {
 
 // ResourceSpec information to identify the resource
 type ResourceSpec struct {
-	Kind      string `json:"kind"`
-	Namespace string `json:"namespace,omitempty"`
-	Name      string `json:"name"`
+	Kind string `json:"kind"`
+	Name string `json:"name"`
```

Comment:
Namespace in CRDs also needs to be removed?

Reply:
Yes,  that was missed. thanks

Reply:
Can we add `Namespace` field back and leave all the other changes there, we still fetch the namespace from matedata.Namespace and leave ResourceSpec.Namespace as empty, and plan to deprecate this a few releases later? One possible failure I could think of is: would it throw an error say namespaces is missing when we fetch/unserialized an old violation?

Reply:
As we are removing a field, we will run into the issue anyways. Question is we remove this now or later. This will be just a dummy field, as since we have rc candidate we can still make the change, as this is related to namespaced policies and not a new feature.

Unless there is another way to deprecate a field more clean way?

Reply:
Even if we leave the namespace in resourceSpec and deprecate gradually? I think this is how Kubernete deprecate things, leave both options available and remove the deprecated one 2 or 3 releases after.

What I suggested was to switch the logic to what this PR does, but keep namespace in resourceSpec for some time, and eventually remove this field. If any third party has integrated with RC version, at least this would not break things.

Reply:
ok, so we'll leave the field in CRD and no use it. And then deprecate in the future.

Reply:
ManagedResourceSpec also uses Namespace, so will add this field back again as well.

Reply:
Sure thanks!